### PR TITLE
Docs: cover custom authentication, rework paginators section

### PR DESCRIPTION
**Pull request by burnash**
Implements #1473

### CODE REVIEW

**Review by VioletM — APPROVED** [review]
Looks good and very clear and structured!

One small thought: Would be useful if in paginator description column there was example of response. This sometimes may even be more useful than just description.

File: docs/website/docs/dlt-ecosystem/verified-sources/rest_api.md [hunk 1]
```diff
@@ -291,46 +291,69 @@ The REST API source will try to automatically handle pagination for you. This wo
...
-This is useful when you're [implementing and using a custom paginator](../../general-usage/http/rest-client.md#custom-paginator).
+:::note
+Currently pagination is supported only for GET requests. To handle POST requests with pagination, you need to implement a [custom paginator](../../general-usage/http/rest-client.md#custom-paginator).
+:::
+
+These are the available paginators:
+
+| `type` | Paginator class | Description |
+| ------------ | -------------- | ----------- |
+| `json_response` | [JSONResponsePaginator](../../general-usage/http/rest-client.md#jsonresponsepaginator) | The link to the next page is in the body (JSON) of the response.<br/>*Parameters:*<ul><li>`next_url_path` (str) - the JSONPath to the next page URL</li></ul> |
```
Comment (VioletM) [inline docs/website/docs/dlt-ecosystem/verified-sources/rest_api.md:348]: The bold may look better here then cursive

```suggestion
| `json_response` | [JSONResponsePaginator](../../general-usage/http/rest-client.md#jsonresponsepaginator) | The link to the next page is in the body (JSON) of the response.<br/>**Parameters:**<ul><li>`next_url_path` (str) - the JSONPath to the next page URL</li></ul> |
```

**Comment by burnash** [comment]
> Would be useful if in paginator description column there was example of response.
Thanks for the comments @VioletM I agree an example would be great, will add it in a follow up PR

File: docs/website/docs/dlt-ecosystem/verified-sources/rest_api.md (hunk 1 above)
Comment (burnash) [reply docs/website/docs/dlt-ecosystem/verified-sources/rest_api.md:348]: @VioletM yes, I'll make it a separate sections as in RESTClient, and use bold font for those.

**Pull request merged**